src/Microsoft.DotNet.Watcher.Core/Internal/Implementation/Project.cs

```diff
@@ -15,12 +17,27 @@ public Project(ProjectModel.Project runtimeProject)
             ProjectFile = runtimeProject.ProjectFilePath;
             ProjectDirectory = runtimeProject.ProjectDirectory;
 
-            Files = runtimeProject.Files.SourceFiles.Concat(
-                    runtimeProject.Files.ResourceFiles.Values.Concat(
-                    runtimeProject.Files.PreprocessSourceFiles.Concat(
-                    runtimeProject.Files.SharedFiles))).Concat(
-                    new string[] { runtimeProject.ProjectFilePath })
-                .ToList();
+            var compilerOptions = runtimeProject.GetCompilerOptions(targetFramework: null, configurationName: null);
+
+            var filesToWatch = new List<string>() { runtimeProject.ProjectFilePath };
+            if (compilerOptions.CompileInclude != null)
+            {
+                filesToWatch.AddRange(compilerOptions.CompileInclude.ResolveFiles());
+
+                if (compilerOptions.EmbedInclude != null)
```

Comment:
> Can you have both of them set at the same time? Does having one set mean ignore the other one?

Yes. The new ones will take priority.

> Which one takes priority because I see that the new properties are null by default if not set but the old ones have default values if not set

When nothing is set, the default of old schema is used.